File: algo/problems/0049-group-anagrams/scala/GroupAnagramsTest.scala
//> using scala 3.4.2
//> using test.dep org.scalameta::munit::1.0.0

class GroupAnagramsTest extends munit.FunSuite:
  private def norm(gs: List[List[String]]): List[List[String]] =
    gs.map(_.sorted).sortBy(_.mkString(","))

  test("basic"):
    val res = GroupAnagrams.groupAnagrams(Array("eat", "tea", "tan", "ate", "nat", "bat"))
    val want = List(List("bat"), List("nat", "tan"), List("ate", "eat", "tea"))
    assertEquals(norm(res), norm(want))

  test("empty string"):
    assertEquals(norm(GroupAnagrams.groupAnagrams(Array(""))), List(List("")))

File: algo/problems/0049-group-anagrams/python/group_anagrams_test.py
import os, sys
sys.path.insert(0, os.path.dirname(__file__))
from group_anagrams import group_anagrams


def _norm(groups):
    return sorted(sorted(g) for g in groups)


def test_basic():
    res = group_anagrams(["eat", "tea", "tan", "ate", "nat", "bat"])
    assert _norm(res) == _norm([["bat"], ["nat", "tan"], ["ate", "eat", "tea"]])

def test_empty_string():
    assert _norm(group_anagrams([""])) == [[""]]

def test_single():
    assert _norm(group_anagrams(["a"])) == [["a"]]
